http-native/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -334,15 +331,13 @@ private static void populateEntityHeaders(BObject messageObj, BObject entity) {
         HttpHeaders httpHeaders = httpCarbonMessage.getHeaders();
         for (String key : httpHeaders.names()) {
             String[] values = httpHeaders.getAll(key).toArray(new String[0]);
-            headers.put(fromString(key.toLowerCase(Locale.getDefault())),
-                        new ArrayValueImpl(io.ballerina.runtime.api.StringUtils.fromStringArray(values)));
+            headers.put(fromString(key.toLowerCase(Locale.getDefault())), fromStringArray(values));
         }
         entity.set(HEADERS_MAP_FIELD, headers);
 
         Set<String> distinctNames = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
         distinctNames.addAll(httpHeaders.names());
-        entity.set(HEADER_NAMES_ARRAY_FIELD, new ArrayValueImpl(
-                io.ballerina.runtime.api.StringUtils.fromStringSet(distinctNames)));
+        entity.set(HEADER_NAMES_ARRAY_FIELD, fromStringSet(distinctNames));
```

Comment:
fromStringSet(distinctNames) we can make this constant since distinctNames is a constant

Reply:
Fixed in #116